pkg/microservice/aslan/core/build/service/build.go

```diff
@@ -394,6 +395,27 @@ func correctFields(build *commonmodels.Build) {
 		build.PostBuild.DockerBuild.DockerFile = strings.Trim(build.PostBuild.DockerBuild.DockerFile, " ")
 		build.PostBuild.DockerBuild.WorkDir = strings.Trim(build.PostBuild.DockerBuild.WorkDir, " ")
 	}
+
+	// modify authType
+	for _, repo := range build.Repos {
+		if repo.Source != setting.SourceFromOther {
+			continue
+		}
+		repo.RepoOwner = strings.TrimPrefix(repo.RepoOwner, "/")
+		repo.RepoOwner = strings.TrimSuffix(repo.RepoOwner, "/")
+		repo.RepoName = strings.TrimPrefix(repo.RepoName, "/")
+		repo.RepoName = strings.TrimSuffix(repo.RepoName, "/")
+		codehosts, err := systemconfig.New().ListCodeHostsInternal()
+		if err != nil {
+			log.Errorf("failed to list codehost,err:%s", err)
+		}
+		for _, codehost := range codehosts {
+			if repo.CodehostID == codehost.ID {
+				repo.AuthType = codehost.AuthType
+				break
+			}
+		}
+	}
```

Comment:
for builds created from build template, repos may be stored in build.Targets[].Repos